cmd/shimdiag/list.go

```diff
@@ -1,25 +1,69 @@
 package main
 
 import (
+	"context"
 	"fmt"
+	"os"
+	"text/tabwriter"
 
 	"github.com/Microsoft/hcsshim/internal/appargs"
+	"github.com/Microsoft/hcsshim/internal/shimdiag"
 	"github.com/urfave/cli"
 )
 
 var listCommand = cli.Command{
 	Name:      "list",
 	Usage:     "Lists running shims",
 	ArgsUsage: " ",
-	Before:    appargs.Validate(),
+	Flags: []cli.Flag{
+		cli.BoolFlag{
+			Name:  "pids",
+			Usage: "Shows the process IDs of each shim",
+		},
+	},
+	Before: appargs.Validate(),
 	Action: func(ctx *cli.Context) error {
+		pids := ctx.Bool("pids")
 		shims, err := findShims("")
 		if err != nil {
 			return err
 		}
+
+		getPid := func(shimStr string) (int32, error) {
+			shim, err := getShim(shimStr)
+			if err != nil {
+				return 0, err
+			}
+			defer shim.Close()
+
+			svc := shimdiag.NewShimDiagClient(shim)
+			resp, err := svc.DiagPid(context.Background(), &shimdiag.PidRequest{})
+			if err != nil {
+				return 0, err
+			}
+			return resp.Pid, nil
+		}
+
+		w := new(tabwriter.Writer)
+		w.Init(os.Stdout, 0, 8, 0, '\t', 0)
+
+		if pids {
+			fmt.Fprintln(w, "Shim \t Pid")
+		}
+
 		for _, shim := range shims {
-			fmt.Println(shim)
+			output := shim
```

Comment:
Why?

Reply:
Shoot, was left in from when it was a flag